torch/_refs/__init__.py

```diff
@@ -4340,9 +4340,13 @@ def linspace(
     computation_dtype, _ = utils.reduction_dtypes(
         rg, REDUCTION_OUTPUT_TYPE_KIND.SAME, dtype_red
     )
-
     cast = partial(torch.full, (), dtype=computation_dtype, **factory_kwargs)
-    out = rg * cast((end - start) / (steps - 1)) + cast(start)
+    # We implement torch.lerp without performing rg / (steps - 1) explicitly
+    # With this we get out[0] == start, out[-1] == end
+    step = cast((end - start) / (steps - 1))
```

Comment:
we shouldn't need to cast `step` to tensor?

Reply:
We need to perform the computation in a higher precision dtype for low-precision dtypes. One way of doing this is by casting `rg` and the other is by casting the numbers to tensors. I chose the latter one, but I could go with the former one if you prefer. I figured that perhaps the latter option only needed to perform manipulations with numbers rather than with the whole array, but the generated code is the same, so I'm fine with either.